File: C++/Types of Matrix/Zero power matrix.cpp
#include <iostream>
#include <vector>
using namespace std;

vector <vector<int>> matrixMultiply(vector <vector<int>> A, vector <vector<int>> B) {
    vector <vector<int>> result (A.size(), vector <int> (B[0].size()));
    int calc;
    for (int row = 0; row < A.size(); row++) {
        for (int col = 0; col < B[row].size(); col++) {
            calc = 0;
            for (int m = 0; m < B.size(); m++) {
                calc += A[row][m] * B[m][col];
            }
            result[row][col] = calc;
        }
    }
    return result;
}
int main(){
    // Zero matrix is just a matrix with any dimensions that has all elements inside the matrix as zero
    vector <vector<int>> A = {
        { 0, 1 },
        { 0, 0 }
    };
    
    if(A.size() == A[0].size()){
        cout << "Matrix: " << endl;
        for (int row = 0; row < A.size(); row++) {
            for (int col = 0; col < A[row].size(); col++) {
                cout << "\t" << A[row][col];
            }
            cout << endl;
        }
        
        vector <vector<int>> result = matrixMultiply(A, A);
        // Print Zero Power matrix after calculation
        cout << "\nZero Power matrix: " << endl;
        bool error = false;
        for (int row = 0; row < result.size(); row++) {
            for (int col = 0; col < result[row].size(); col++) {
                cout << "\t" << result[row][col];
                if(result[row][col] != 0) error = true;
            }
            cout << endl;
        }
        if(error) cout << "=> Not a valid zero power matrix.";
    }else{
        cout << "Error! Multiplication not possible" << endl;
    }
}

File: Python/Types of Matrix/Zero power matrix.py
def matrixMultiply(A, B):
    result = [[None]*len(B[0]) for row in range(len(A))]
    for row in range(0, len(A)):
        for col in range(0, len(B[row])):
            calc = 0
            for m in range(0, len(B)):
                calc += A[row][m] * B[m][col]
            result[row][col] = calc
    return result


# Zero matrix is just a matrix with any dimensions that has all elements inside the matrix as zero
A = [[0, 1],
     [0, 0]]

if len(A) == len(A[0]):
    print("Matrix: ")
    for row in range(0, len(A)):
        for col in range(0, len(A[row])):
            print("\t", A[row][col], end='')
        print()
    
    result = matrixMultiply(A, A)

    # Print Zero Power matrix after calculation
    print("\nZero Power matrix: ")
    error = False
    for row in range(0, len(result)):
        for col in range(0, len(result[row])):
            print("\t", result[row][col], end='')
            if result[row][col] != 0: error = True
        print()
    if error: print("=> Not a valid zero power matrix.")
else:
    print("Error! Multiplication not possible")